### PR TITLE
Fix nested one command code flow control

### LINKED ISSUE #255: Nested one command code flow control + Workaround
### Description

Some commands cause the next line to be indented. For example:

```autohotkey
if (true)
    MsgBox
Loop % n
    SoundBeep
```

Such code will be well formatted.

However, the formatter doesn't work with nested commands.

### Unformatted input snippet

```autohotkey
if (true)
if (true)
MsgBox
MsgBox
```

### Expected formatted output snippet

```autohotkey
if (true)
    if (true)
        MsgBox
MsgBox
```

### Actual formatted output snippet

```autohotkey
if (true)
    if (true)
    MsgBox
MsgBox
```

### Workaround

As a workaround, use braces:

```autohotkey
if (true) {
    if (true)
        MsgBox
}
MsgBox
```

### PR DESCRIPTION
**Pull request by kyklish**
Closes #255.

Fix bug in PR #260: continuation section in nested objects initialization.

This PR and PR #260 makes single-line comments jump +-1 indent. Revise single-line comments by explicitly aligning single-line comments to the next line of code after the main code has been formatted. In addition, this reduces possible formatting errors when we encounter blank lines or single-line comments in the file.

Extract oneCommandCode detection to util function `nextLineIsOneCommandCode(...)`

Notifying @mark-wiemer

### CODE REVIEW

**Comment by mark-wiemer** [comment]
This is a big PR! I'll review it soon enough. My main goal right now is getting AHK+ updated to match AHK++ :)

File: src/providers/formattingProvider.utils.ts [hunk 1]
```diff
@@ -474,3 +522,158 @@ export function replaceAll(
...
+                // trim formatter directive comments, they saved with indentation
+                line.trim(),
+                options.preserveIndent,
+            );
+        } else {
+            depth = calculateDepth(line, options);
+            // obj := { a: { b: 1
+            //         ; comment
+            //         , c: 2 }} <-- don't match close braces in object's
+            //                                              continuation section
```
Comment (mark-wiemer) [inline src/providers/formattingProvider.utils.ts:652]:
```suggestion
            //                       continuation section
```

File: src/test/suite/format/samples/255-nested-one-command-code.in.ahk [hunk 2]
```diff
@@ -0,0 +1,407 @@
+; [Issue #255](https://github.com/mark-wiemer/vscode-autohotkey-plus-plus/issues/255)
+if {
```
Comment (mark-wiemer) [inline src/test/suite/format/samples/255-nested-one-command-code.in.ahk:2]: Can we split this into multiple tests? 400+ lines is huge, we should aim for less than 20 lines per sample file if possible.

Each file can have the `255` prefix still, no reason there can't be more than one test file for a bug :)

File: src/test/suite/providers/formatting/formattingProvider.utils.test.ts [hunk 3]
```diff
@@ -655,6 +659,278 @@ suite('FormattingProvider utils', () => {
...
+        // List of test data
+        let dataList = [
+            // {
+            //     in: , // input test string
+            //     op: , // formatting options
+            //     rs: , // expected result
+            // },
+            {
+                in: '',
+                op: { insertSpaces: true, tabSize: 4, preserveIndent: true },
```
Comment (mark-wiemer) [inline src/test/suite/providers/formatting/formattingProvider.utils.test.ts:800]: For `op`, let's use the `defaultArgs` pattern introduced with this code. https://github.com/mark-wiemer/vscode-autohotkey-plus-plus/blob/29403f32a54d9526e1d824f0876372f6a0dbceaa/src/test/suite/service/runnerService.test.ts#L7

**Review by mark-wiemer — CHANGES_REQUESTED** [review]

File: src/test/suite/providers/formatting/formattingProvider.utils.test.ts (hunk 3 above)
Comment (kyklish) [reply src/test/suite/providers/formatting/formattingProvider.utils.test.ts:800]: I not understand, why you wrote twice same parameters in declaration `defaultArgs`:

```TypeScript
const defaultArgs: {
    compilePath: 'mockCompilePath';
} = {
    compilePath: 'mockCompilePath',
}
```

Why not like this?

```TypeScript
const defaultArgs: {
    compilePath: string;
} = {
    compilePath: 'mockCompilePath',
}
```

File: src/test/suite/format/samples/255-nested-one-command-code.in.ahk (hunk 2 above)
Comment (kyklish) [reply src/test/suite/format/samples/255-nested-one-command-code.in.ahk]: Done.

File: src/test/suite/providers/formatting/formattingProvider.utils.test.ts (hunk 3 above)
Comment (kyklish) [reply src/test/suite/providers/formatting/formattingProvider.utils.test.ts]: Done with `defaultFormattingOptions`.

**Pull request merged**